src/test/java/org/mockito/internal/creation/proxy/ProxyMockMakerTest.java

```diff
@@ -0,0 +1,51 @@
+/*
+ * Copyright (c) 2021 Mockito contributors
+ * This program is made available under the terms of the MIT License.
+ */
+package org.mockito.internal.creation.proxy;
+
+import org.junit.Test;
+import org.mockito.internal.creation.AbstractMockMakerTest;
+
+import java.io.Serializable;
+import java.lang.reflect.Proxy;
+
+import static org.assertj.core.api.Assertions.assertThat;
+
+public class ProxyMockMakerTest
```

Comment:
Can we maybe also have some tests that verify that attempting to mock non-interfaces correctly fails with a descriptive error message?

Reply:
Based on https://app.codecov.io/gh/mockito/mockito/compare/2397/diff#D2 I think it would also be good to add some tests that method invocations and all are working as expected.

Reply:
We test for the `mockable` state. To avoid the overhead, those methods will never be invoked with non-mockable types and the error messages are rather undefined.